pkg/clients/database/configuration/mysql.go

```diff
@@ -0,0 +1,118 @@
+// https://github.com/Azure/azure-sdk-for-go/blob/master/services/mysql/mgmt/2017-12-01/mysql/models.go
+// https://github.com/Azure/azure-sdk-for-go/blob/master/services/postgresql/mgmt/2017-12-01/postgresql/models.go
+
+// MySQLConfigurationAPI represents the API interface for a MySQL Server Configuration client
+type MySQLConfigurationAPI interface {
+	Get(ctx context.Context, s *azuredbv1beta1.MySQLServerConfiguration) (mysql.Configuration, error)
+	CreateOrUpdate(ctx context.Context, s *azuredbv1beta1.MySQLServerConfiguration) error
+	Delete(ctx context.Context, s *azuredbv1beta1.MySQLServerConfiguration) error
+	GetRESTClient() autorest.Sender
+}
+
+// MySQLConfigurationClient is the concreate implementation of the MySQLConfigurationAPI interface for MySQL that calls Azure API.
+type MySQLConfigurationClient struct {
+	mysql.ConfigurationsClient
+}
+
+// NewMySQLConfigurationClient creates and initializes a MySQLConfigurationClient instance.
+func NewMySQLConfigurationClient(cl mysql.ConfigurationsClient) *MySQLConfigurationClient {
+	return &MySQLConfigurationClient{
+		ConfigurationsClient: cl,
+	}
+}
+
+// GetRESTClient returns the underlying REST client that the client object uses.
+func (c *MySQLConfigurationClient) GetRESTClient() autorest.Sender {
+	return c.ConfigurationsClient.Client
+}
+
+// Get retrieves the requested MySQL Configuration
+func (c *MySQLConfigurationClient) Get(ctx context.Context, cr *azuredbv1beta1.MySQLServerConfiguration) (mysql.Configuration, error) {
+	return c.ConfigurationsClient.Get(ctx, cr.Spec.ForProvider.ResourceGroupName, cr.Spec.ForProvider.ServerName, cr.Spec.ForProvider.Name)
+}
+
+// CreateOrUpdate creates or updates a MySQL Server Configuration
+func (c *MySQLConfigurationClient) CreateOrUpdate(ctx context.Context, cr *azuredbv1beta1.MySQLServerConfiguration) error {
+	return c.update(ctx, cr, cr.Spec.ForProvider.Value, nil)
+}
+
+// Delete deletes the given MySQL Server Configuration
+func (c *MySQLConfigurationClient) Delete(ctx context.Context, cr *azuredbv1beta1.MySQLServerConfiguration) error {
+	source := SourceSystemManaged
+	// we are mimicking Terraform behavior here: when the configuration object
+	// is deleted, we are resetting its value to the system default,
+	// and updating its source to "system-default" to declare that
+	// we are no longer managing it.
+	return c.update(ctx, cr, &cr.Status.AtProvider.DefaultValue, &source)
+}
+
+func (c *MySQLConfigurationClient) update(ctx context.Context, cr *azuredbv1beta1.MySQLServerConfiguration, value, source *string) error {
+	s := cr.Spec.ForProvider
+	config := mysql.Configuration{
+		ConfigurationProperties: &mysql.ConfigurationProperties{
+			Value:  value,
+			Source: source,
+		},
+	}
+	op, err := c.ConfigurationsClient.CreateOrUpdate(ctx, s.ResourceGroupName, cr.Spec.ForProvider.ServerName, cr.Spec.ForProvider.Name, config)
+	if err != nil {
+		return err
```

Comment:
We can wrap this SDK error with `errors.wrap`.

Reply:
We return this error with wrap in other functions where the function is called. I'm not sure we need to return with wrap here again. Other functions in the file also return without wrap.

Reply:
As a practice, I try to make sure that any errors that are returned from 3rd party libs are always wrapped. And then I do not wrap errors that are returned from our functions, because we make sure that they already wrap errors they return. We also assume that new errors are instantiated with associated stack traces, e.g., with `errors.New`.